modules/bmaas-setting-up-a-dedicated-namespace.adoc

```diff
@@ -0,0 +1,172 @@
+----
++
+For example, when using the `htpasswd` identity provider, run the following command:
++
+[source,terminal]
+----
+$ oc create secret generic htpass-secret \
+  --from-file=htpasswd=users.htpasswd -n openshift-config
+----
+
+. Configure OAuth to use the identity provider:
+
+.. Edit the OAuth resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit oauth cluster
+----
++
+The editor opens and displays the Oauth resource.
+
+.. Add the identity provider configuration to the `spec.identityProviders` list:
++
+.Identity provider configuration examples
+[options="header"]
+|====
+|Type|Example
+| htpasswd
+a|[source,yaml]
+----
+# ...
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: htpasswd
+  htpasswd:
+    fileData:
+      name: <secret>
+# ...
+----
+| LDAP
+a|[source,yaml]
+----
+# ...
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: ldap
+  ldap:
+    attributes:
+      id:
+      - dn
+      email:
+      - mail
+      name:
+      - cn
+      preferredUsername:
+      - uid
+# ...
+----
+| GitHub
+a|[source,yaml]
+----
+# ...
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: GitHub
+    github:
+      ca:
+        name: ca-config-map
+      clientID: {...}
+      clientSecret:
+        name: github-secret
+      hostname: ...
+      organizations:
+      - myorganization1
+      - myorganization2
+      teams:
+      - myorganization1/team-a
+      - myorganization2/team-b
+# ...
+----
+|====
++
+For more information about identify providers, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/{product-version}/html-single/authentication_and_authorization/index[Authentication and authorization].
+
+.. Save and exit the editor.
+
+. Create a BMaaS `bmadmin` user by running the following command:
++
+[source,terminal]
+----
+$ oc create user <username>
+----
+<username>::
+The user name. Replace `<username>` with your username. The following examples use `bmadmin` as the username.
+
+. Create a BMaaS `bmadmin` user with the cluster's identity provider. For example, when using the `htpasswd` identify provider, run the following command:
++
+[source,terminal]
+----
+$ htpasswd -c -B -b ./users.htpasswd <username> <password>
+----
+<username>::
+The `htpasswd` username. Replace `<username>` with the same username from the previous step.
+<password>::
+The `htpasswd` password.
+
+. Create a dedicated `bmaas` namespace for BMaaS hosts by running the following command:
++
+[source,terminal]
+----
+$ oc new-project <namespace>
+----
+`<namespace>`::
+Replace <namespace> with the namespace name that you want to use.
+
+. Assign the `edit` role to the BMaaS `bmadmin` user in the `bmaas` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-user edit <username> -n bmaas
+----
+
+. Login as the `bmadmin` user by running the following command:
++
+[source,terminal]
+----
+$ oc login <api_server_url>:6443
```

Comment:
This should be the last step after cluster-admin applies the need roles to user  `bmadmin` and before the creation of BMC secret